components/brave_extension/extension/brave_extension/background/api/shieldsAPI.ts

```diff
@@ -164,23 +164,32 @@ export const setAllowScriptOriginsOnce = (origins: Array<string>, tabId: number)
 
 export type GetViewPreferencesData = {
   showAdvancedView: boolean
+  statsBadgeVisible: boolean
 }
 
 const settingsKeys = {
-  showAdvancedView: { key: 'brave.shields.advanced_view_enabled', type: chrome.settingsPrivate.PrefType.BOOLEAN }
+  showAdvancedView: { key: 'brave.shields.advanced_view_enabled', type: chrome.settingsPrivate.PrefType.BOOLEAN },
+  statsBadgeVisible: { key: 'brave.shields.stats_badge_visible', type: chrome.settingsPrivate.PrefType.BOOLEAN },
 }
 export async function getViewPreferences (): Promise<GetViewPreferencesData> {
   const showAdvancedViewPref = await SettingsPrivate.getPreference(settingsKeys.showAdvancedView.key)
+  const statsBadgeVisiblePref = await SettingsPrivate.getPreference(settingsKeys.statsBadgeVisible.key)
```

Comment:
We can use `promise.all` here instead of waiting for each setting. As an extra, since we store the neccessary config in one object, this code also lends itself to avoiding repetition with something like:

```js
let settings = {}
await Promise.all(Object.keys(settingsKeys).map(async (name) => {
  // Get setting by internal key
  const pref = await SettingsPrivate.getPreference(settingsKeys[name].key)
  // Validate setting type
  if (pref.type !== settingsKeys[name].type) {
    throw new Error(`Unexpected settings type received for "${settingsKeys[name].key}". Expected: ${settingsKeys[name].type}, Received: ${pref.type}`)
  }
  // Valid
  settings[name] = pref.value
})
return settings
```

wdyt?

Reply:
sounds good, updated with the above

Reply:
that one is tricky because change happens with shields closed so there's no `state` since the component isn't mounted. I moved the data to the persistent state (shields localStorage). lmk if you have a better option